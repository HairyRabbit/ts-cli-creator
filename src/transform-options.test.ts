import { Project, ts, printNode } from 'ts-morph'
import convert from './transform-option'

describe(`convert() types`, () => {
  test(`string`, () => {
    const code: string = `interface Options { foo: string }`
    expect(print(code)).toMatchSnapshot()
  })
  
  test(`number`, () => {
    const code: string = `interface Options { foo: number }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`boolean`, () => {
    const code: string = `interface Options { foo: boolean }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`string[]`, () => {
    const code: string = `interface Options { foo: string[] }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`Array<string>`, () => {
    const code: string = `interface Options { foo: Array<string> }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`number[]`, () => {
    const code: string = `interface Options { foo: number[] }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`Array<number>`, () => {
    const code: string = `interface Options { foo: Array<number> }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`boolean[]`, () => {
    const code: string = `interface Options { foo: boolean[] }`
    expect(print(code)).toMatchSnapshot()
  })

  test(`Array<boolean>`, () => {
    const code: string = `interface Options { foo: Array<boolean> }`
    expect(print(code)).toMatchSnapshot()
  })

  test.only(`enum`, () => {
    const code: string = `\
enum E { A = 'a', B = 'b' }
interface Options { foo: E }
`
    expect(print(code)).toMatchSnapshot()
  })
})

function print(code: string): string {
  const project = new Project({
    skipFileDependencyResolution: true
  })
  const sourceFile = project.createSourceFile(`tmp.ts`, code)
  const nodes = convert(sourceFile.getInterfaces()[0])
  // console.log(require('util').inspect(nodes[0], { depth: null }))
  console.log(printNode(nodes[0]))
  return ts.createPrinter().printList(ts.ListFormat.MultiLine, nodes as any, sourceFile as any)
}
