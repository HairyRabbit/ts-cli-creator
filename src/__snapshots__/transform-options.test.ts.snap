// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert() types Array<boolean> 1`] = `
"option(\\"foo\\", {
    type: \\"boolean\\",
    array: true
})
"
`;

exports[`convert() types Array<number> 1`] = `
"option(\\"foo\\", {
    type: \\"number\\",
    array: true
})
"
`;

exports[`convert() types Array<string> 1`] = `
"option(\\"foo\\", {
    type: \\"string\\",
    array: true
})
"
`;

exports[`convert() types boolean 1`] = `
"option(\\"foo\\", {
    type: \\"boolean\\"
})
"
`;

exports[`convert() types boolean[] 1`] = `
"option(\\"foo\\", {
    type: \\"boolean\\",
    array: true
})
"
`;

exports[`convert() types enum 1`] = `
"option(\\"foo\\", {
    type: \\"string\\",
    choices: [E.A, E.B]
})
"
`;

exports[`convert() types number 1`] = `
"option(\\"foo\\", {
    type: \\"number\\"
})
"
`;

exports[`convert() types number[] 1`] = `
"option(\\"foo\\", {
    type: \\"number\\",
    array: true
})
"
`;

exports[`convert() types string 1`] = `
"option(\\"foo\\", {
    type: \\"string\\"
})
"
`;

exports[`convert() types string[] 1`] = `
"option(\\"foo\\", {
    type: \\"string\\",
    array: true
})
"
`;
